include/unifex/let_value_with.hpp

```diff
@@ -100,20 +102,18 @@ template<typename StateFactory, typename SuccessorFactory, typename Receiver>
 struct _operation<StateFactory, SuccessorFactory, Receiver>::type {
   template <typename StateFactory2, typename SuccessorFactory2, typename Receiver2>
   type(StateFactory2&& stateFactory, SuccessorFactory2&& func, Receiver2&& r) :
-    stateFactory_(static_cast<StateFactory2&&>(stateFactory)),
     func_(static_cast<SuccessorFactory2&&>(func)),
-    state_(static_cast<StateFactory&&>(stateFactory_)()),
+    state_(std::forward<StateFactory2>(stateFactory)()),
```

Comment:
The `stateFactory_` member is not actually used after the constructor, so I removed it.

Reply:
It _is_ used in the sense that the `stateFactory_` might be stateful (e.g. a lambda with captures). You'll need to put this back.

Reply:
I should have realized the code was not accidentally written that way :P